lib/SampleService/core/config.py

```diff
@@ -207,15 +231,46 @@ def get_validators(url: str) -> MetadataValidatorSet:
     :returns: A set of metadata validators.
     '''
     # TODO VALIDATOR make validator CLI
+
     try:
-        with _urllib.request.urlopen(url) as res:
-            cfg = _yaml.safe_load(res)
+        config_asset = None
+        if url:
+            config_url = url
+        elif not repo_path:
+            raise ValueError(f'No metadata validator config URL or repo path.')
+        else:
+            try: 
+                repo = _Github(login_or_token=token).get_repo(repo_path)
+                releases = [rel for rel in repo.get_releases() if prerelease_ok or not rel.prerelease]
+            except:
+                raise RuntimeError(f'Fetching releases from repo {repo_path} failed.')
+            if not releases:
+                raise ValueError(f'No releases found in validator config repo {repo_path}')
+            latest_release = releases[0] # max(releases, key=lambda rel: rel.created_at)
+            assets = latest_release.get_assets()
+            if not assets:
+                raise ValueError(f'No assets found in validator config repo {repo_path}')
+            config_asset = next((a for a in assets if a.name==repo_file), None)
+            if not config_asset:
+                raise ValueError(f'No config asset found in validator config repo {repo_path}')
+            config_url = config_asset.url
+
+        req = request.Request(config_url)
+        req.add_header('Accept', 'application/octet-stream')
+        if token:
+            req.add_header('Authorization', f'token {token}')
+        with request.urlopen(req) as response:
+            cfg = _yaml.safe_load(response)
+
     except _URLError as e:
-        raise ValueError(
-            f'Failed to open validator configuration file at {url}: {str(e.reason)}') from e
+        if config_asset:
+            raise ValueError(f'Error downloading config asset from {config_asset.url}: {str(e.reason)}') from e
+        else:
+            raise ValueError(f'Error downloading config asset from {url or repo_path}: {str(e.reason)}') from e
     except _ParserError as e:
         raise ValueError(
-            f'Failed to open validator configuration file at {url}: {str(e)}') from e
+            f'Failed to open validator configuration file from {url or repo_path}: {str(e)}') from e
```

Comment:
Same re relevant info for repo_path

Reply:
Same as above. I don't think `config_asset == True` necessarily means that there's no repo path, right? Or maybe I don't quite grok the logic

Reply:
It does mean that, yes. If a url is provided, it isnt loaded via a repo asset (config_asset) and doesn't use the repo_path logic (see lines 236-241)

Reply:
yep, you're right after reading from the top rather than skimming like a lazy chump

I kind of think the method is too long now (~60 lines) but I leave it up to you if you want to do anything about it

Reply:
:+1: